public/javascripts/ScrabBag.js

```diff
@@ -0,0 +1,54 @@
+class ScrabBag {
```

Comment:
After additional reading of the code, I also note a lot of interdependency between the functions - they are tightly coupled, and seem to require a specific order of execution to work properly.  A client consuming this class would need to be intimately familiar with the internal implementation details of this class in order to properly order function calls to ensure correct execution.  Narrowing the API, as described above, forces clients to use this class in an appropriate manner.